### PR TITLE
Modularize petab objective

### PR DESCRIPTION
**Pull request by yannikschaelte**
split into

* parameter mapping
* edatas construction
* parameter overridin in edatas

which can easily be reused in pypesto

### CODE REVIEW

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ python/amici/petab_objective.py  -11
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5186803&bid=17017605)

File: python/amici/petab_objective.py [hunk 1]
```diff
@@ -133,72 +124,50 @@ def simulate_petab(
...
-
-    :param simulation_conditions:
-        Result of petab.get_simulation_conditions. Can be provided to save
-        time if this has be obtained before.
-
-    :param parameter_mapping:
-        Optional precomputed PEtab parameter mapping for efficiency.
-
+        Run simulation with these parameters. If None, PEtab `nominalValues`
+        will be used). To be provided as dict,  mapping PEtab problem
```
Comment (dweindl) [inline python/amici/petab_objective.py]:
```suggestion
        will be used). To be provided as dict, mapping PEtab problem
```

File: python/amici/petab_objective.py [hunk 2]
```diff
@@ -133,72 +124,50 @@ def simulate_petab(
...
+        parameters to SBML IDs.
     :param scaled_parameters:
         If True, problem_parameters are assumed to be on the scale provided
-        in the PEtab parameter table and will be unscaled. If False,
-        they are assumed to be in linear scale.
+        in the PEtab parameter table and will be unscaled. If False, they
+        are assumed to be in linear scale.
+    :param parameter_mapping:
+        Optional precomputed PEtab parameter mapping for efficiency.
```
Comment (dweindl) [inline python/amici/petab_objective.py]: Has to be from create_parameter_mapping, right?

File: python/amici/petab_objective.py [hunk 3]
```diff
@@ -133,72 +124,50 @@ def simulate_petab(
...
+    :param simulation_conditions:
+        Result of `petab.get_simulation_conditions`. Can be provided to save
+        time if this has be obtained before.
```
Comment (dweindl) [inline python/amici/petab_objective.py]:
```suggestion
        time if this has been obtained before.
```

File: python/amici/petab_objective.py [hunk 4]
```diff
@@ -209,115 +178,108 @@ def edatas_from_petab(
...
-    :param observable_ids:
-        List of observable IDs
-
-    :param parameter_mapping:
-        PEtab parameter mapping for current condition
-
+        PEtab problem
+    :param simulation_conditions:
+        Result of `petab.get_simulation_conditions`. Can be provided to save
+        time if this has be obtained before.
```
Comment (dweindl) [inline python/amici/petab_objective.py]:
```suggestion
        time if this has been obtained before.
```

File: python/amici/petab_objective.py [hunk 5]
```diff
@@ -466,15 +390,204 @@ def _set_initial_concentration(condition_id, species_id, init_par_id,
...
+
+    :return:
+        ExpData instance.
+    """
+    # extract measurement table rows for condition
+    measurement_df = petab.get_rows_for_condition(
+        measurement_df=petab_problem.measurement_df, condition=condition)
+
+    if amici_model.nytrue != len(observable_ids):
+        raise AssertionError("Number of AMICI model observables does not"
```
Comment (dweindl) [inline python/amici/petab_objective.py]:
```suggestion
        raise AssertionError("Number of AMICI model observables does not "
```

**Comment by yannikschaelte** [comment]
> Bit hard to follow the diff. Tests pass so seems to be fine :)

yes, maybe look at the files next to each other better.

File: python/amici/petab_objective.py (hunk 2 above)
Comment (yannikschaelte) [reply python/amici/petab_objective.py]: yes, will document.

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ python/amici/petab_objective.py  -11
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=5186803&bid=17017605)
